src/stratigraphy/evaluation/evaluation_dataclasses.py

```diff
@@ -0,0 +1,138 @@
+            dict: The object as a dictionary.
+        """
+        return {
+            f"{feature_name}_precision": self.precision,
+            f"{feature_name}_recall": self.recall,
+            f"{feature_name}_f1": self.f1,
+        }
+
+    @staticmethod
+    def from_metric_list(metric_list: list["Metrics"]) -> "Metrics":
+        """Converts a list of metrics to a metric.
+
+        Args:
+            metric_list (list): The list of metrics.
+
+        Returns:
+            Metrics: Combined metrics.
+        """
+        tp = sum([metric.tp for metric in metric_list])
+        fp = sum([metric.fp for metric in metric_list])
+        fn = sum([metric.fn for metric in metric_list])
+        return Metrics(tp=tp, fp=fp, fn=fn)
+
+
+@dataclass
+class BoreholeMetadataMetrics(metaclass=abc.ABCMeta):
+    """Metrics for metadata."""
+
+    elevation_metrics: Metrics
+    coordinates_metrics: Metrics
+
+    def to_json(self) -> dict:
+        """Converts the object to a dictionary.
+
+        Returns:
+            dict: The object as a dictionary.
+        """
+        return {
+            **self.elevation_metrics.to_json("elevation"),
+            **self.coordinates_metrics.to_json("coordinate"),
+        }
+
+
+@dataclass
+class FileBoreholeMetadataMetrics(BoreholeMetadataMetrics):
+    """Single file Metrics for borehole metadata."""
+
+    filename: str
+
+    def get_document_level_metrics(self) -> pd.DataFrame:
+        """Get the document level metrics."""
+        return pd.DataFrame(
+            data={
+                "document_name": [self.filename],
+                "elevation": [self.elevation_metrics.f1],
+                "coordinate": [self.coordinates_metrics.f1],
+            }
+        )
+
+
+@dataclass
+class OverallBoreholeMetadataMetrics(metaclass=abc.ABCMeta):
+    """Metrics for borehole metadata."""
+
+    borehole_metadata_metrics: list[BoreholeMetadataMetrics] = None
+
+    def __init__(self):
+        """Initializes the FileBoreholeMetadataMetrics object."""
+        self.borehole_metadata_metrics = []
+
+    def get_cumulated_metrics(self) -> dict:
+        """Evaluate the metadata metrics."""
+        elevation_metrics = Metrics.from_metric_list(
+            [metadata.elevation_metrics for metadata in self.borehole_metadata_metrics]
+        )
+        coordinates_metrics = Metrics.from_metric_list(
+            [metadata.coordinates_metrics for metadata in self.borehole_metadata_metrics]
+        )
+        return BoreholeMetadataMetrics(
+            elevation_metrics=elevation_metrics, coordinates_metrics=coordinates_metrics
+        ).to_json()
+
+    def get_document_level_metrics(self) -> pd.DataFrame:
+        """Get the document level metrics."""
+        # Collect all the data frames in a list
+        frames = [metadata.get_document_level_metrics() for metadata in self.borehole_metadata_metrics]
+
+        # Concatenate them once at the end
+        return pd.concat(frames, ignore_index=True)
```

Comment:
This does not return the input documents in the original order. Where does the order get lost?

Maybe the implementation approach from `DatasetMetricsCatalog.document_level_metrics_df` is more robust?

Reply:
I cannot reproduce the issues you are mentioning. 

I tried something along these lines: 

```python

        document_level_metrics = pd.DataFrame(columns=["document_name", "elevation", "coordinate"])
        for metadata in self.borehole_metadata_metrics:
            document_level_metrics = document_level_metrics.merge(metadata.get_document_level_metrics(), how="outer")

```

But this actually sorted the filenames, and the order was lost. Is there a reason why the other is important here in your opinion?

Reply:
I initially had this implementation 

```
document_level_metrics = pd.DataFrame(columns=["document_name", "elevation", "coordinate"])
for metadata in self.borehole_metadata_metrics:
    document_level_metrics = pd.concat([document_level_metrics, metadata.get_document_level_metrics()])
```

but this raises a deprecation warning.

Reply:
I see. I already suggested a new fix. Maybe you can have a look at it. There is even an assertion that makes sure the files are in order.

Reply:
You assertion ensures that the order is unchanged (i.e. execution order), but this is not necessarily alphabetical, at least on my system it isn't.